common/hooks/pre-pkg/06-shlib-provides.sh

```diff
@@ -42,6 +42,39 @@ collect_sonames() {
 	rm -f ${_tmpfile}
 }
 
+# verify that shlibs listed in common/shlibs are actually provided by the package
+verify_sonames() {
+	local _destdir="$1" broken= mapshlibs pkgshlibs mappedshlibs
+	mapshlibs="${XBPS_COMMONDIR}/shlibs"
+
+	if [ ! -f ${_destdir}/shlib-provides ]; then
+		return 0
+	fi
+
+	pkgshlibs="$(cat ${_destdir}/shlib-provides)"
+
+	mappedshlibs="$(grep -E "[[:blank:]]${pkgname}" $mapshlibs)" || return 0
+
+	set -- $mappedshlibs
+	while [ $# -gt 0 ]
+	do
+		if [ "$($XBPS_UHELPER_CMD getpkgname "$2")" = ${pkgname} ]; then
+			if [[ "${pkgshlibs}" != *"$1"* ]]; then
+				msg_red "shlib '$1' isn't in shlib-provides for ${pkgname}\n"
+				broken=1
+			fi
+
+			file="$(find ${_destdir} -name "$1")"
+			if [ -z "$file" ]; then
+				msg_red "file for shlib '$1' isn't in ${pkgname}\n"
+			fi
+		fi
+		shift 2
```

Comment:
There is third optional column in shlibs. 99-pkglint.sh use  `grep shlibs -e ... | while read conflictFile conflictPkg ignore; do ...` to handle that. And then interpretation of third column is needed for glibc.